### PR TITLE
Merge bitcoin#14291, #14275, #14465, #14559, #14561

### PR DESCRIPTION
**Pull request by 5tefan**

### CODE REVIEW

**Comment by 5tefan** [comment]
It's odd that linux64-test failed, but linux32-test passed in CI. It's passing on my machine. Random occurrence that doesn't matter (can someone rerun CI?), or maybe some kind of race condition?

File: src/wallet/walletutil.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 #include <wallet/walletutil.h>
+#include <util/system.h>
```
Comment (PastaPastaPasta) [inline src/wallet/walletutil.cpp:6]: 14291 should be a newline between this and walletutil.h include

File: src/wallet/walletutil.h [hunk 2]
```diff
@@ -7,10 +7,16 @@
 
 #include <chainparamsbase.h>
 #include <util/system.h>
```
Comment (PastaPastaPasta) [inline src/wallet/walletutil.h:9]: 14291 these should be dropped

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Comment by PastaPastaPasta** [comment]
Otherwise, looks good :+1:

**Comment by 5tefan** [comment]
Thanks. Changes to 14291 done.

File: test/functional/wallet_multiwallet.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def wallet_file(name):
         wallet_names = ['w1', 'w2', 'w3', 'w', 'sub/w5', os.path.join(self.options.tmpdir, 'extern/w6'), 'w7_symlink', 'w8', '']
         extra_args = ['-wallet={}'.format(n) for n in wallet_names]
         self.start_node(0, extra_args)
-        assert_equal(set(map(lambda w: w['name'], self.nodes[0].listwalletdir()['wallets'])), set(['', 'w3', 'w2', 'sub/w5', 'w7', 'w7', 'w1', 'w8', 'w']))
+        assert_equal(sorted(map(lambda w: w['name'], self.nodes[0].listwalletdir()['wallets'])), ['', os.path.join('sub', 'w5'), 'w', 'w1', 'w2', 'w3', 'w7', 'w7_symlink', 'w8'])
```
Comment (UdjinM6) [inline test/functional/wallet_multiwallet.py:71]: 14561 requires 14559 for this to be backported cleanly

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/wallet_multiwallet.py (hunk 3 above)
Comment (5tefan) [reply test/functional/wallet_multiwallet.py:71]: Interesting, I see 14559 introduced `os.path.join`. Just added 14559 before 14561.  Curious that `wallet_names` doesn't use `os.path.join` and instead hardcodes `/` char.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**